base/show.jl

```diff
@@ -2516,9 +2524,58 @@ function show_tuple_as_call(io::IO, name::Symbol, sig::Type;
     end
     print_within_stacktrace(io, ")", bold=true)
     show_method_params(io, tv)
+    if get(out, :limit, false)::Bool
+        sz = get(out, :displaysize, (typemax(Int), typemax(Int)))::Tuple{Int, Int}
+        str = String(take!(unwrapcontext(io)[1]))
+        print(out, depth_limit(str, sz[2]))
+    end
     nothing
 end
 
+function depth_limit(str::String, n::Int)
+    depth = 0
+    width_at = Int[]
+    strwid = 0
+    in_escape = false
+    for c in str
+        if c == '}'
+            depth -= 1
+        elseif c == '\e'
+            in_escape = true
+        end
+        in_escape || (strwid += textwidth(c))
+        if depth > 0
+            if depth > length(width_at)
+                resize!(width_at, depth)
+                width_at[depth] = 0
+            end
+            in_escape || (width_at[depth] += textwidth(c))
+        end
+        if c == '{'
+            depth += 1
+        elseif in_escape && c == 'm'
+            in_escape = false
+        end
+    end
```

Comment:
This would fail on a type such as:
```
julia> Val{'{'}
Val{'{'}
```

Reply:
This might be better after https://github.com/JuliaLang/julia/pull/49586, since making this reliable and general is the point of that PR IIUC (similar to https://github.com/JuliaLang/julia/pull/27430 which also had that, but which I never managed to finish)